chain/network-primitives/src/config.rs

```diff
@@ -13,6 +13,7 @@ pub struct NetworkConfig {
     pub account_id: Option<AccountId>,
     pub addr: Option<SocketAddr>,
     pub boot_nodes: Vec<PeerInfo>,
+    pub whitelist_nodes: Vec<PeerInfo>,
```

Comment:
I used whitelist_nodes instead of whitelist, because it shares string format with boot_nodes. I've added a private type WhitelistNode ~ (PeerId, SocketAddr, Option<AccountId>) and added "parsing" PeerInfo -> WhitelistNode to the constructor of PeerManagerActor. I've removed the debug_asserts.